src/librustc_resolve/diagnostics.rs

```diff
@@ -397,6 +397,76 @@ impl Bar {
 ```
 "##,
 
+E0417: r##"
+A static variable was referenced in a pattern. Example of erroneous code:
+
+```
+static FOO : i32 = 0;
+
+match 0 {
+    FOO => {} // error: static variables cannot be referenced in a
+              //        pattern, use a `const` instead
+    _ => {}
+}
+```
+
+Compiler needs to know the pattern value at compile's time, which is
+not possible with a `static` variable. So please verify you didn't
```

Comment:
Please verify that the variable is spelled correctly and if so, try to use a `const` instead.